onnxruntime/core/framework/session_state.cc

```diff
@@ -228,7 +227,7 @@ Status SessionState::GeneratePatternGroupCache(const std::vector<std::reference_
     auto* node = graph_viewer_->GetNode(node_plan.node_index);
     int output_start = node_index + static_cast<int>(node->InputDefs().size()) + static_cast<int>(node->ImplicitInputDefs().size());
     //allocate output
-    for (int i = 0; i < static_cast<int>(node->OutputDefs().size()); ++i) {
+    for (int i = 0, end = static_cast<int>(node->OutputDefs().size()); i < end; ++i) {
```

Comment:
can we just use size_t and avoid the cast to int here?

Reply:
I think so, let me change that too